crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -210,43 +210,75 @@ fn assert_component_access_compatibility(
            query_type, filter_type, system_name, accesses);
 }
 
-/// Use to safely access and interact with up to 8 mutually exclusive [`SystemParam`]s such as
-/// two queries that reference the same mutable data  or a mutable query and acess to the ecs world.
+/// A collecton of potentially conflicting [`SystemParam`]s allowed by disjoint access.
 ///
-/// Access to each individual [`SystemParam`] by using the funcions ```p0()```, ```p1()```, ..., ```p7()```. This ensures that
-/// there's either only one mutable reference to a parameter at a time or an infinite* number of inmutable references.
-/// # Example
-/// ```
-/// use bevy_ecs::prelude::*;
+/// Allows to safely access and interact with up to 8 mutually exclusive [`SystemParam`]s such as
+/// two queries that reference the same mutable data  or a mutable query and acess to the ecs world.
 ///
-/// #[derive(Component)]
-/// struct Health(u32);
+/// Access to each individual [`SystemParam`] by using the functions `p0()`, `p1()`, ..., `p7()`. This ensures that
+/// there's either only one mutable reference to a parameter at a time or any number of immutable references.
 ///
-/// #[derive(Component)]
-/// struct Enemy;
+/// # Example
 ///
-/// //Given the following system
+/// ```
+/// # use bevy_ecs::prelude::*;
+/// #
+/// # #[derive(Component)]
+/// # struct Health;
+/// #
+/// # #[derive(Component)]
+/// # struct Enemy;
+/// #
+/// # #[derive(Component)]
+/// # struct Ally;
+/// #
+/// // Given the following system
 /// fn fancy_system(
 ///     mut set: ParamSet<(
 ///         Query<&mut Health, With<Enemy>>,
-///         Query<&mut Health, Without<Enemy>>,
+///         Query<&mut Health, With<Ally>>,
 ///         &World
```

Comment:
```suggestion
///         &World,
```

It is a  convention to put a comma even at the last function parameter when they are listed vertically. The main reason for this is to ease the versioning process in case of adding or removing a parameter from the list (one line diff instead of two).